docs/rfcs/008-more-access-control-oidc-oauth.md

```diff
@@ -0,0 +1,95 @@
+# Open Id Connect / OAuth
+
+[RFC 005-Access-Control](005-Access-Control.md) outlines how we can use a RoleProvider service to _authenticate_ users and allow the DLCS to _authorize_ their access to image resources.
+
+This document follows on from this to look at how we can use oauth2/oidc as an alternative method of role provisioning.
+
+## DLCS as Oauth 2.0 Client
+
+The preference would be for the DLCS to be a direct _Client_ of an OAuth2.0 _Authorization Server_, rather than going via a _role-provider-like_ middleman using the [`Authorization Code Flow`](https://oauth.net/2/grant-types/authorization-code/):
+
+![Authorization Code Flow](sequence-src/auth-oauth.png "Authorization Code Flow")
+
+The goal is to get an `id_token` with custom claims that the DLCS can then map to DLCS-specific roles via configuration.
+
+### Configuration
+
+The DLCS `"roleprovider"` database record (not to be confused with RoleProvider application) contains 2 relevant columns:
+
+* `"configuration"` which defines how to log a user in, log them out and get roles etc.
+* `"credentials"` which contains any auth credentials required in using above configuration.
+
+#### Role Provider - CAS
+
+An example of the currently supported `"configuration"` block for `"roleprovider"`, to use with a RoleProvider application is:
+
+```json
+{
+  "default": {
+    "config": "cas",
+    "target": "http://roleprovider.example.dlcs/login",
+    "roles": "http://roleprovider.example.dlcs/roles",
+    "logout": "http://roleprovider.example.dlcs/logout"
+  }
+}
+```
+
+Where:
+* `<key>` is either `"default"`, or a specific hostname. If the current host matches the hostname then that config block is used, else it falls back to `"default"`.
+* `config` - The type of configuration. Currently only `"cas"` is supported.
+* `target` - Where the DLCS should redirect user to login. On successful login it sends a token back to DLCS.
+* `roles` - URL to POST token to to fetch current user roles (`"credentials"` in the form `{"username": "xxx", "password": "xxx"}` supported to use as basic-auth credentials).
+* `logout` - URL to log user out and end their session.
+
+#### Role Provider - OAuth2
+
+An alternative configuration block could be (this example uses values from the Wellcome [auth-test](https://github.com/wellcomecollection/iiif-builder/tree/master/src/AuthTest) application):
+
+```json
+{
+    "default": {
+        "config": "oauth2",
+        "provider": "auth0",
+        "domain": "<domain>.eu.auth0.com",
```

Comment:
Being pedantic here, but I would suggest that documentation such as this shouldn't imply usage of a specific authorisation server such as Auth0.

Reply:
Agreed - removed in favour of `"<domain>"` only.